internal/security/client/vaultclient.go

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/common"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+
+	"github.com/edgexfoundry/go-mod-secrets/pkg"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/providers/vault"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/token/authtokenloader"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/token/fileioperformer"
+)
+
+// Vault is the structure to get the secret client from go-mod-secrets vault package
+type Vault struct {
```

Comment:
Why is this not part of go-mod-secrets?

Reply:
This is a wrapper to use go-mod-secrets. it is part of go-mod-bootstrap and I think we can change it to use it from go-mod-boostrap directly for this.

Reply:
discussed with @lenny-intel  and this can be part of future stories on abstracting the secret provider into go-mod-secrets later.